tests/integration/test_split_command.py

```diff
@@ -0,0 +1,136 @@
+import shutil
+from pathlib import Path
+
+import pytest
+import yaml
+from click.testing import CliRunner
+
+from dbt_meshify.dbt import Dbt
+from dbt_meshify.main import split
+
+src_project_path = "test-projects/split/split_proj"
+dest_project_path = "test-projects/split/temp_proj"
+
+
+def setup_test_project():
+    src_path = Path(src_project_path)
+    dest_path = Path(dest_project_path)
+    shutil.copytree(src_path, dest_path)
+
+
+def teardown_test_project():
+    dest_path = Path(dest_project_path)
+    shutil.rmtree(dest_path)
+
+
+class TestSplitCommand:
+    def test_split_one_model(self):
+        setup_test_project()
+        runner = CliRunner()
+        result = runner.invoke(
+            split,
+            [
+                "my_new_project",
+                "--project-path",
+                dest_project_path,
+                "--select",
+                "stg_orders",
+            ],
+        )
+
+        assert result.exit_code == 0
+        assert (
+            Path(dest_project_path) / "my_new_project" / "models" / "staging" / "stg_orders.sql"
+        ).exists()
+        x_proj_ref = "{{ ref('my_new_project', 'stg_orders') }}"
+        child_sql = (Path(dest_project_path) / "models" / "marts" / "orders.sql").read_text()
+        assert x_proj_ref in child_sql
+        teardown_test_project()
```

Comment:
Would a teardown after an `assert` work as expected? I'd think that it would stop at the `assert` but it might just be my lack of `pytest` knowledge.

Reply:
it seems to work locally!